cmd/agentrun/agentrun/cli/main.py

```diff
@@ -0,0 +1,514 @@
+"""
+Main CLI entry point for AgentRun.
+
+This module defines the command-line interface using Typer, providing
+a rich and developer-friendly experience for managing AI agents.
+"""
+
+from pathlib import Path
+from typing import Optional, List
+
+import typer
+from rich.console import Console
+from rich.progress import Progress, SpinnerColumn, TextColumn
+from rich.table import Table
+
+from agentrun.runtime.build_runtime import BuildRuntime
+from agentrun.runtime.invoke_runtime import InvokeRuntime
+from agentrun.runtime.pack_runtime import PackRuntime
+from agentrun.runtime.publish_runtime import PublishRuntime
+from agentrun.runtime.status_runtime import StatusRuntime
+
+# Initialize rich console for beautiful output
+console = Console()
+
+# Create the main Typer application
+app = typer.Typer(
+    name="agentrun",
+    help="AgentRun CLI - A developer tool for packaging, building, and deploying AI agents to AgentCube",
+    no_args_is_help=True,
+    rich_markup_mode="rich",
+    add_completion=False,
+)
+
+# Version callback
+def version_callback(value: bool) -> None:
+    """Show version information and exit."""
+    if value:
+        from agentrun import __version__
+        console.print(f"AgentRun CLI (kubectl agentrun) version: [bold green]{__version__}[/bold green]")
+        raise typer.Exit()
+
+# Error handling  
+def _handle_error(e: Exception, command_name: str, verbose: bool):
+    console.print(f"Error {command_name}: [red]{str(e)}[/red]")
+    if verbose:
+        import traceback
+        console.print(traceback.format_exc())
+    raise typer.Exit(1)
+
+@app.callback()
+def main(
+    version: Optional[bool] = typer.Option(
+        None,
+        "--version",
+        "-v",
+        help="Show version and exit",
+        callback=version_callback,
+        is_eager=True,
+    ),
+    verbose: bool = typer.Option(
+        False,
+        "--verbose",
```

Comment:
should accept "-v", and other conventional shortcuts, such as "-p" for "--port",  “-ep” for "endpoint", etc.